packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -352,69 +321,61 @@ export interface DeploymentsProps {
 }
 
 /**
- * SDK obtained by assuming the deploy role
- * for a given environment
+ * Scope for a single set of deployments from a set of Cloud Assembly Artifacts
+ *
+ * Manages lookup of SDKs, Bootstrap stacks, etc.
  */
-export interface PreparedSdkForEnvironment {
-  /**
-   * The SDK for the given environment
-   */
-  readonly stackSdk: ISDK;
+export class Deployments {
+  public readonly envs: EnvironmentAccess;
 
   /**
-   * The resolved environment for the stack
-   * (no more 'unknown-account/unknown-region')
-   */
-  readonly resolvedEnvironment: cxapi.Environment;
-  /**
-   * The Execution Role that should be passed to CloudFormation.
+   * SDK provider for asset publishing (do not use for anything else).
+   *
+   * This SDK provider is only allowed to be used for that purpose, nothing else.
    *
-   * @default - no execution role is used
+   * It's not a different object, but the field name should imply that this
+   * object should not be used directly, except to pass to asset handling routines.
    */
-  readonly cloudFormationRoleArn?: string;
+  private readonly assetSdkProvider: SdkProvider;
 
   /**
-   * Access class for environmental resources to help the deployment
+   * SDK provider for passing to deployStack
+   *
+   * This SDK provider is only allowed to be used for that purpose, nothing else.
+   *
+   * It's not a different object, but the field name should imply that this
+   * object should not be used directly, except to pass to `deployStack`.
    */
-  readonly envResources: EnvironmentResources;
-}
+  private readonly deployStackSdkProvider: SdkProvider;
 
-/**
- * Scope for a single set of deployments from a set of Cloud Assembly Artifacts
- *
- * Manages lookup of SDKs, Bootstrap stacks, etc.
- */
-export class Deployments {
-  private readonly sdkProvider: SdkProvider;
-  private readonly sdkCache = new Map<string, SdkForEnvironment>();
   private readonly publisherCache = new Map<AssetManifest, cdk_assets.AssetPublishing>();
-  private readonly environmentResources: EnvironmentResourcesRegistry;
 
   private _allowCrossAccountAssetPublishing: boolean | undefined;
   constructor(private readonly props: DeploymentsProps) {
-    this.sdkProvider = props.sdkProvider;
-    this.environmentResources = new EnvironmentResourcesRegistry(props.toolkitStackName);
+    this.assetSdkProvider = props.sdkProvider;
+    this.deployStackSdkProvider = props.sdkProvider;
```

Comment:
If they are the same and both are private props, why split them?

Reply:
To indicate for what purpose they may be used. If you are writing code for "look up bootstrap stack", you will not use the `assetSdkProvider` nor the `deployStackSdkProvider`.

Reply:
👍🏻 With the descriptions that makes sense.